jsx/src/Store.js

```diff
@@ -0,0 +1,28 @@
+export const initialState = {
+  user_data: undefined,
+  user_page: 0,
+  groups_data: undefined,
+  groups_page: 0,
+  limit: 50,
```

Comment:
Is this configurable? We have about 15k users in our JupyterHub sometimes, and paginating through them is impossible since search doesn't really exist. Right now, the only way to do this is to have all 15k show in the page at once and use the browser's search feature.

Reply:
Configuration of the react app is a bigger question. I typically inject the setting in a json in the HTML header and let them read by the react app.

Regarding the many users case, would a typeahead component like http://ericgio.github.io/react-bootstrap-typeahead/ be a good answer, at least better than a  15k lines table?

Reply:
This could do the job. @yuvipanda do you need that feature (the typeahead, or the injected settings) before merging?

<img width="896" alt="Screenshot 2021-05-18 at 09 35 34" src="https://user-images.githubusercontent.com/226720/118610876-894a1380-b7bc-11eb-84d9-eb9657eb8445.png">

Reply:
@echarles yeah, I think some way to search for users would be important I think. Right now this can be accomplished by setting a huge limit and using browser's search functionality. Some way to accomplish that would be required I think.

We can also just set a limit in the URL. But not sure how the app will react to 15k elements rendered :D

Reply:
So to recap, there are 2 options:

1. Add a json in the html header (in the jinja template) populated from a new setting like `c.JupyterHub.max_row_per_page``
2. Add that 3rd party component that allows to search

I let @naatebarber elaborate on more options and give feedback to see if he can pull that requirement in this PR.

> We can also just set a limit in the URL. But not sure how the app will react to 15k elements rendered :D

Quite badly I expect.

Reply:
@yuvipanda @echarles This is a good point. Currently on the API side we have a pagination default limit (initially 50) and a pagination maximum limit (200). The default limit can be changed with the jupyterhub_config, but if we're dealing with 15k+ users and 200 item-per-page maximums I feel like we'd need some back-end search functionality. 

The best thing we have right now is the `/users/{username}` endpoint, which at least lets us know if a user exists. However I don't think it'd be too difficult to add another endpoint, or at least expand on the one above, to provide some basic text-matching of usernames. Not sure if it's out of scope for this PR, but if it's done it'll definitely be a more efficient way to handle a bunch of users like you guys are saying.

Even if it is out of scope, I think it'd be cool to add search endpoints later on to pull back users and groups by their various attributes. Thoughts?

Reply:
@echarles 
 > Add a json in the html header (in the jinja template) populated from a new setting like `c.JupyterHub.max_row_per_page``
Good call btw, we should be passing the default limit / custom limit set in the config to the front end